core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -77,6 +80,17 @@ private Deletes() {
     return filter.filter(rows);
   }
 
+  public static <T> CloseableIterable<T> match(CloseableIterable<T> rows,
```

Comment:
`match` is not a good candidate for me to express the meaning of finding the existing row data that hits the equality delete sets. I may need a better name for this.